python/cuml/test/test_sgd.py

```diff
@@ -2,28 +2,66 @@
 import cudf
 from cuml.solvers import SGD as cumlSGD
 import pytest
+from sklearn.datasets.samples_generator import make_blobs
+from sklearn import datasets
+import pandas as pd
 
 
 @pytest.mark.parametrize('lrate', ['constant', 'invscaling', 'adaptive'])
 @pytest.mark.parametrize('datatype', [np.float32, np.float64])
 @pytest.mark.parametrize('input_type', ['dataframe', 'ndarray'])
 @pytest.mark.parametrize('penalty', ['none', 'l1', 'l2', 'elasticnet'])
 @pytest.mark.parametrize('loss', ['hinge', 'log', 'squared_loss'])
-def test_svd(datatype, lrate, input_type, penalty, loss):
-    X = np.array([[-1, -1], [-2, -1], [1, 1], [2, 1]], dtype=datatype)
-    y = np.array([1, 1, 2, 2], dtype=datatype)
-    pred_data = np.array([[3.0, 5.0], [2.0, 5.0]]).astype(datatype)
-    if input_type == 'dataframe':
-        X = cudf.DataFrame()
-        X['col1'] = np.asarray([-1, -2, 1, 2], dtype=datatype)
-        X['col2'] = np.asarray([-1, -1, 2, 2], dtype=datatype)
-        y = cudf.Series(np.array(y, dtype=np.float32))
-        pred_data = cudf.DataFrame()
-        pred_data['col1'] = np.asarray([3, 2], dtype=datatype)
-        pred_data['col2'] = np.asarray([5, 5], dtype=datatype)
+@pytest.mark.parametrize('name', [pytest.param('None', marks=pytest.mark.unit),
+                                  pytest.param('blobs',
+                                               marks=pytest.mark.stress),
+                                  pytest.param('iris',
+                                               marks=pytest.mark.quality)])
+def test_svd(datatype, lrate, input_type, penalty,
+             loss, name):
+
+    if name == 'blobs':
+        n_samples = 500000
+        train_rows = np.int32(n_samples*0.8)
+        X, y = make_blobs(n_samples=n_samples,
+                          n_features=1000, random_state=0)
+        X_test = np.array(X[train_rows:, 0:]).astype(datatype)
+        X_train = np.array(X[0:train_rows, :]).astype(datatype)
+        y_train = np.array(y[0:train_rows, ]).astype(datatype)
+
+    elif name == 'iris':
+        iris = datasets.load_iris()
+        X = iris.data
+        y = iris.target
+        train_rows = np.int32((np.shape(X)[0])*0.8)
+        X_test = np.array(X[train_rows:, 0:]).astype(datatype)
```

Comment:
Just a little stylistic thing... maybe a better practice to do:  `np.array(foo, dtype=datatype)` rather than constructing and then changing-type. All in one call that way.

Reply:
changed :)